apps/aecore/lib/aecore/peers/scheduler.ex

```diff
@@ -19,9 +19,11 @@ defmodule Aecore.Peers.Scheduler do
     Peers.check_peers()
     Sync.introduce_variety()
     Sync.refill()
-    Sync.ask_peers_for_unknown_blocks(Peers.all_peers())
-    Sync.add_valid_peer_blocks_to_chain()
+    if(Peers.is_chain_synced?()) do
```

Comment:
why the if?

Reply:
This way we don't queue any unnecessary casts (only called if we have finished syncing)

Reply:
Ok, I didn't notice

Reply:
`Peers.is_chain_synced?()` returns false if our top block height is lower then the one reported by peers. `ask_peers_for_unknown_blocks` download's the blocks we don't know. So now we will never get our chain synced as `ask_peers_for_unknown_blocks` that's exactly what `ask_peers_for_unknown_blocks` does.

Reply:
You are right, the thing I had in mind was to check if a sync is in progress, maybe we can have a status variable for the sync module and check that instead.